ui/decredmaterial/switch.go

```diff
@@ -63,68 +63,46 @@ func (s *SwitchButtonText) Layout(gtx layout.Context) layout.Dimensions {
 	card.Radius = CornerRadius{NE: 8, NW: 8, SE: 8, SW: 8}
 	return card.Layout(gtx, func(gtx C) D {
 		return layout.UniformInset(unit.Dp(2)).Layout(gtx, func(gtx C) D {
-			return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
-				layout.Rigid(func(gtx C) D {
-					return layout.Stack{}.Layout(gtx,
-						layout.Stacked(func(gtx C) D {
-							return s.activeCard.Layout(gtx, func(gtx C) D {
-								return layout.Inset{
-									Left:   m8,
-									Bottom: m4,
-									Right:  m8,
-									Top:    m4,
-								}.Layout(gtx, func(gtx C) D {
-									txt := s.t.Body2(s.activeTxt)
-									txt.Color = s.activeTextColor
-									if !s.isActive {
-										txt.Color = s.inactiveTextColor
-									}
-									return txt.Layout(gtx)
-								})
-							})
-						}),
-						layout.Expanded(s.activeBtn.Layout),
+			list := &layout.List{Axis: layout.Horizontal}
+			Items := s.items[1:]
+			return list.Layout(gtx, len(Items), func(gtx C, i int) D {
+				return layout.UniformInset(unit.Dp(0)).Layout(gtx, func(gtx C) D {
+					index := i + 1
+					btn := s.items[index].button
+					btn.Inset = layout.Inset{
+						Left:   m8,
+						Bottom: m4,
+						Right:  m8,
+						Top:    m4,
+					}
+					return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+						layout.Rigid(btn.Layout),
 					)
-				}),
-				layout.Rigid(func(gtx C) D {
-					return layout.Stack{}.Layout(gtx,
-						layout.Stacked(func(gtx C) D {
-							return s.inactiveCard.Layout(gtx, func(gtx C) D {
-								return layout.Inset{
-									Left:   m8,
-									Bottom: m4,
-									Right:  m8,
-									Top:    m4,
-								}.Layout(gtx, func(gtx C) D {
-									txt := s.t.Body2(s.inactivetxt)
-									txt.Color = s.activeTextColor
-									if !s.isInactive {
-										txt.Color = s.inactiveTextColor
-									}
-									return txt.Layout(gtx)
-								})
-							})
-						}),
-						layout.Expanded(s.inactiveBtn.Layout),
-					)
-				}),
-			)
+				})
+			})
 		})
 	})
 }
 
 func (s *SwitchButtonText) handleClickEvent() {
-	for s.inactiveBtn.Clicked() {
-		s.inactiveCard.Color = s.t.Color.Surface
-		s.activeCard.Color = color.NRGBA{}
-		s.isActive = false
-		s.isInactive = true
-	}
+	for i := range s.items {
```

Comment:
I dont follow. I also indicate the index. Kindly elaborate.

Reply:
you have i as the index, then reassign it to a more convenient naming of `index` on the next line
i was pointing out that it might be better to just rename `i` to `index` directly